### PR TITLE
65 improve config

### LINKED ISSUE #65: Improve config
Improve configuration workflow. 
* Fall back to broker config if none was set.
* Create default config on first start up
* Expose more config variables.

### PR DESCRIPTION
**Pull request by swenzel**
closes #65

### CODE REVIEW

File: esque/cli/commands.py [hunk 1]
```diff
@@ -142,6 +141,21 @@ def config_autocomplete():
     )
 
 
+@config.command("edit", help="Edit your esque config file.")
+@error_handler
+def config_edit():
```
Comment (hfjn) [inline esque/cli/commands.py]: This is awesome. ❤️

File: esque/config/__init__.py [hunk 2]
```diff
@@ -27,91 +29,69 @@
...
-        return [f"{host_name}:{self.bootstrap_port}" for host_name in self.bootstrap_hosts]
+    def bootstrap_servers(self) -> List[str]:
+        return self.current_context_dict["bootstrap_servers"]
 
     @property
-    def default_partitions(self) -> int:
-        config_dict = self.current_context_dict
-        return int(config_dict["default_partitions"])
+    def default_num_partitions(self) -> int:
+        return self.default_values.get("num_partitions", 1)
```
Comment (hfjn) [inline esque/config/__init__.py]: Can we get those values from broker config?

File: esque/config/__init__.py [hunk 3]
```diff
@@ -198,40 +185,40 @@ def _get_confluent_ssl_config(self) -> Dict[str, str]:
             return {}
         rdk_conf = {}
         if self.current_context_dict.get("ssl_cafile"):
-            rdk_conf["ssl.ca.location"] = self.current_context_dict["ssl_cafile"]
+            rdk_conf["ssl.ca.location"] = self.ssl_params["cafile"]
         if self.current_context_dict.get("ssl_certfile"):
```
Comment (garrettthomaskth) [inline esque/config/__init__.py]:
```suggestion
        if self.ssl_params.get("certfile"):
```

File: esque/config/__init__.py [hunk 4]
```diff
@@ -198,40 +185,40 @@ def _get_confluent_ssl_config(self) -> Dict[str, str]:
...
-            rdk_conf["ssl.certificate.location"] = self.current_context_dict["ssl_certfile"]
+            rdk_conf["ssl.certificate.location"] = self.ssl_params["certfile"]
         if self.current_context_dict.get("ssl_keyfile"):
```
Comment (garrettthomaskth) [inline esque/config/__init__.py]:
```suggestion
        if self.ssl_params.get("keyfile"):
```

File: esque/config/__init__.py [hunk 5]
```diff
@@ -198,40 +185,40 @@ def _get_confluent_ssl_config(self) -> Dict[str, str]:
             return {}
         rdk_conf = {}
         if self.current_context_dict.get("ssl_cafile"):
```
Comment (garrettthomaskth) [inline esque/config/__init__.py]: if self.ssl_params.get("cafile"):

File: esque/config/__init__.py (hunk 3 above)
Comment (garrettthomaskth) [inline esque/config/__init__.py]: if self.ssl_params.get("certfile"):

File: esque/config/__init__.py (hunk 4 above)
Comment (garrettthomaskth) [inline esque/config/__init__.py]: if self.ssl_params.get("keyfile"):

File: esque/config/__init__.py [hunk 6]
```diff
@@ -198,40 +185,40 @@ def _get_confluent_ssl_config(self) -> Dict[str, str]:
...
-            rdk_conf["ssl.ca.location"] = self.current_context_dict["ssl_cafile"]
+            rdk_conf["ssl.ca.location"] = self.ssl_params["cafile"]
         if self.current_context_dict.get("ssl_certfile"):
-            rdk_conf["ssl.certificate.location"] = self.current_context_dict["ssl_certfile"]
+            rdk_conf["ssl.certificate.location"] = self.ssl_params["certfile"]
         if self.current_context_dict.get("ssl_keyfile"):
-            rdk_conf["ssl.key.location"] = self.current_context_dict["ssl_keyfile"]
+            rdk_conf["ssl.key.location"] = self.ssl_params["keyfile"]
         if self.current_context_dict.get("ssl_password"):
```
Comment (garrettthomaskth) [inline esque/config/__init__.py]: if self.ssl_params.get("password"):

File: esque/config/sample_config.yaml [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+  # This is an example configuration for connecting to an SSL enabled
+  # endpoint
+  ssl_enabled:
+    bootstrap_servers:
+      - localhost:9092
+    # esque can infer the protocol SSL if ssl_params are given
+    # security_protocol: SSL
+
+    # the following parameters are optioal
```
Comment (garrettthomaskth) [inline esque/config/sample_config.yaml]:
```suggestion
    # the following parameters are optional
```

File: esque/config/__init__.py (hunk 5 above)
Comment (garrettthomaskth) [reply esque/config/__init__.py]: sorry, I would have "suggested changes" but github was doing something weird with the indenting, so I just commented

File: esque/config/__init__.py [hunk 8]
```diff
@@ -27,114 +29,100 @@
...
-        config_dict = self.current_context_dict
-        return int(config_dict["default_partitions"])
+    def default_num_partitions(self) -> int:
+        return self.default_values.get("num_partitions", 1)
 
     @property
     def default_replication_factor(self) -> int:
-        config_dict = self.current_context_dict
-        return int(config_dict["default_replication_factor"])
+        return self.default_values.get("replication_factor", min(len(self.bootstrap_servers), 3))
```
Comment (hfjn) [inline esque/config/__init__.py]: Same here

**Review by hfjn — APPROVED** [review]
LGTM. Some two small comments. :)

**Pull request merged**